File: bankingOnGolang/internal/model/FeeCharge.go
package model

import (
	"time"
)

// ==============================================================
// FeeCharge Declaration
// ==============================================================
type FeeCharge struct {
	BaseModel
	FeeCode       string
	Amount        Money
	AppliedOn     time.Time
	AccountId     *uint
	Account       *Account `gorm:"foreignKey:AccountId"`
	LoanAccountId *uint
	LoanAccount   *LoanAccount `gorm:"foreignKey:LoanAccountId"`
	FeeType       FeeType

	// parent associations as their child

}

File: src/app/models/FeeCharge.ts


// Define collection and schema for FeeCharge
export  FeeCharge {
    feeCode: string
    amount: Money
    appliedOn: Date
    Account: Schema.Types.ObjectId
    LoanAccount: Schema.Types.ObjectId
    FeeType:  String
#
    collection: 'feeCharges'
}
